tstest/integration/vms/vms_test.go

```diff
@@ -94,6 +95,19 @@ func fetchDistro(t *testing.T, resultDistro Distro) {
 	qcowPath := filepath.Join(cdir, "qcow2", resultDistro.sha256sum)
 
 	_, err = os.Stat(qcowPath)
+	if err == nil {
+		hash := checkCachedImageHash(t, resultDistro, cdir)
+		if hash != resultDistro.sha256sum {
+			t.Logf("hash for %s (%s) doesn't match expected %s, re-downloading", resultDistro.name, qcowPath, resultDistro.sha256sum)
+		}
+
+		if err := os.Remove(qcowPath); err != nil {
+			t.Fatalf("[unexpected] can't delete wrong cached image: %v", err)
+		}
+
+		err = errors.New("sentinel value to force redownload")
```

Comment:
this is not quite a sentinel if it doesn't have a named variable that others == on.

maybe "some non-nil error"